--- a/src/BOops.cpp	2026-08-30 21:58:58.998016450 +0700
+++ b/src/BOops.cpp	2026-08-31 03:52:42.049337694 +0700
@@ -22,6 +22,7 @@
 #include <string>
 #include <stdexcept>
 #include <algorithm>
+#include <vector>
 #include "BOops.hpp"
 #include "ControllerLimits.hpp"
 #include "BUtilities/stof.hpp"
@@ -1613,7 +1614,8 @@
 
 	// Store pads
 	{
-		char padDataString[0x8010] = "\nMatrix data:\n";
+		std::vector<char> padDataString (0x8010, '\0');
+		std::copy_n ("\nMatrix data:\n", sizeof ("\nMatrix data:\n"), padDataString.begin ());
 
 		for (int pgNr = 0; pgNr <= pageMax; ++pgNr)
 		{
@@ -1630,26 +1632,26 @@
 						snprintf (valueString, 62, "pg:%d; sl:%d; st:%d; gt:%1.3f; sz:%d; mx:%1.3f", pgNr, slotNr, stepNr, p.gate, int (p.size), p.mix);
 						if ((slotNr < NR_SLOTS - 1) || (stepNr < NR_STEPS)) strcat (valueString, ";\n");
 						else strcat(valueString, "\n");
-						strcat (padDataString, valueString);
+						strcat (padDataString.data (), valueString);
 					}
 
 				}
 			}
 		}
-		store (handle, urids.bOops_statePad, padDataString, strlen (padDataString) + 1, urids.atom_String, LV2_STATE_IS_POD);
+		store (handle, urids.bOops_statePad, padDataString.data (), strlen (padDataString.data ()) + 1, urids.atom_String, LV2_STATE_IS_POD);
 	}
 
 	// Store Keys
 	{
-		char keysDataString[8192] = "";
+		std::vector<char> keysDataString (8192, '\0');
 
 		for (int pageNr = 0; pageNr <= pageMax; ++pageNr)
 		{
-			strcat (keysDataString, "\nKeys data slots page ");
+			strcat (keysDataString.data (), "\nKeys data slots page ");
 			char pageString[16];
 			snprintf (pageString, 12, "%d", pageNr);
-			strcat (keysDataString, pageString);
-			strcat (keysDataString, ":\n");
+			strcat (keysDataString.data (), pageString);
+			strcat (keysDataString.data (), ":\n");
 			for (int slotNr = 0; slotNr < NR_SLOTS; ++slotNr)
 			{
 				if (pages[pageNr].keys[slotNr][NR_PIANO_KEYS])
@@ -1659,29 +1661,29 @@
 					char keysString[40];
 					bool2hstr<std::array<bool, NR_PIANO_KEYS + 1>> (pages[pageNr].keys[slotNr], keysString);
 
-					strcat (keysDataString, slotString);
-					strcat (keysDataString, " key: 0x");
-					strcat (keysDataString, keysString);
-					strcat (keysDataString, ";\n");
+					strcat (keysDataString.data (), slotString);
+					strcat (keysDataString.data (), " key: 0x");
+					strcat (keysDataString.data (), keysString);
+					strcat (keysDataString.data (), ";\n");
 				}
 			}
 		}
 
-		store (handle, urids.bOops_keysData, keysDataString, strlen (keysDataString) + 1, urids.atom_String, LV2_STATE_IS_POD);
+		store (handle, urids.bOops_keysData, keysDataString.data (), strlen (keysDataString.data ()) + 1, urids.atom_String, LV2_STATE_IS_POD);
 	}
 
 	// Store shapes
 	{
-		char shapesDataString[0x110000] = "";
+		std::vector<char> shapesDataString (0x110000, '\0');
 
 		// Slot shapes
 		for (int pageNr = 0; pageNr <= pageMax; ++pageNr)
 		{
-			strcat (shapesDataString, "\nShape data slots page ");
+			strcat (shapesDataString.data (), "\nShape data slots page ");
 			char pageString[16];
 			snprintf (pageString, 12, "%d", pageNr);
-			strcat (shapesDataString, pageString);
-			strcat (shapesDataString, ":\n");
+			strcat (shapesDataString.data (), pageString);
+			strcat (shapesDataString.data (), ":\n");
 			for (int slotNr = 0; slotNr < NR_SLOTS; ++slotNr)
 			{
 				if (pages[pageNr].shapes[slotNr] != Shape<SHAPE_MAXNODES>())
@@ -1706,16 +1708,16 @@
 						);
 						if ((slotNr < NR_SLOTS - 1) || (nodeNr < pages[pageNr].shapes[slotNr].size ())) strcat (valueString, ";\n");
 						else strcat(valueString, "\n");
-						strcat (shapesDataString, valueString);
+						strcat (shapesDataString.data (), valueString);
 					}
 				}
 			}
-			strcat(shapesDataString, "\n");
+			strcat(shapesDataString.data (), "\n");
 		}
 
 
 		// Param shapes
-		strcat (shapesDataString, "\nShape data param:\n");
+		strcat (shapesDataString.data (), "\nShape data param:\n");
 		for (int slotNr = 0; slotNr < NR_SLOTS; ++slotNr)
 		{
 			if (!slots[slotNr].shape.isDefault())
@@ -1740,11 +1742,11 @@
 					);
 					if ((slotNr < NR_SLOTS - 1) || (nodeNr < slots[slotNr].shape.size ())) strcat (valueString, ";\n");
 					else strcat(valueString, "\n");
-					strcat (shapesDataString, valueString);
+					strcat (shapesDataString.data (), valueString);
 				}
 			}
 		}
-		store (handle, urids.bOops_shapeData, shapesDataString, strlen (shapesDataString) + 1, urids.atom_String, LV2_STATE_IS_POD);
+		store (handle, urids.bOops_shapeData, shapesDataString.data (), strlen (shapesDataString.data ()) + 1, urids.atom_String, LV2_STATE_IS_POD);
 	}
 
 	return LV2_STATE_SUCCESS;
